### PR TITLE
fix: verify version download

### PR DESCRIPTION
**Pull request by catsby**
fix for https://github.com/defenseunicorns/setup-uds/issues/11 by verifying the success of downloading the release, and error if not.

### CODE REVIEW

File: action.yaml [hunk 1]
```diff
@@ -53,7 +53,11 @@ runs:
         esac
 
         echo "Downloading UDS CLI ${version} for ${{ runner.os }} ${arch}"
-        curl -sL "${release_path}/uds-cli_${version}_${filename}" -o "${target_dir}/uds"
+        curl -f -sL "${release_path}/uds-cli_${version}_${filename}" -o "${target_dir}/uds"
+        if [ $? -ne 0 ]; then
```
Comment (mjnagel) [inline action.yaml:57]: If this curl fails it exits the script and won't get to the next line to output this error message, see https://github.com/defenseunicorns/setup-uds/actions/runs/12380598836/job/34557339290?pr=12#step:3:59

File: .github/workflows/setup-uds.yaml [hunk 2]
```diff
@@ -50,3 +50,29 @@ jobs:
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: Setup UDS CLI with defaults
+        id: error_step
+        uses: ./
+        with:
+          # renovate: datasource=github-tags depName=defenseunicorns/uds-cli versioning=semver
+          version: v0.01.9
```
[the remark this answers is not in the document]
Comment (catsby) [reply .github/workflows/setup-uds.yaml]: whoops, that's bad/hasty copy-pasta, fixed in bf47625

File: action.yaml (hunk 1 above)
Comment (catsby) [reply action.yaml]: Updated in a5432bc

**Review by mjnagel — APPROVED** [review]

**Review by decleaver — APPROVED** [review]
when @mjnagel says its approved, its approved

**Pull request merged**